src/logging.rs

```diff
@@ -0,0 +1,115 @@
+use colored::*;
+use env_logger::Builder;
+use indicatif::ProgressBar;
+use log::{Level, LevelFilter};
+use std::{
+    fmt,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc, RwLock,
+    },
+};
+
+static MAX_MODULE_WIDTH: AtomicUsize = AtomicUsize::new(0);
```

Comment:
What is the purpose of this variable?

For me it is not clear what module means here.

Reply:
I renamed this to use a better name. Basically it is remembering how wide the window is to try and adjust the padding of all the logs.
I took this functionality from pretty_env_logger :)